wordpress-mobile/WordPress-iOS
# My Site Dashboard (Phase 2): Quick Start Initial Structure

Part of #17877

## Description
- Adds the initial structure for the Quick Start dashboard card
- Triggers Quick Start from the dashboard view
- Displays a site menu waypoint if the tour is initiated from the dashboard view (if needed)

<img src="https://user-images.githubusercontent.com/6711616/155394130-c2a640a6-eb1a-44fc-a402-85cfc3fedb3a.png" width=200>

## Notes
- TODO: Refine Quick Start dashboard card design
- TODO: Show Quick Start on Site Menu or Dashboard (one or the other, not both)

## How to test

Make sure the MSD feature flag is enabled

1. First we need to reset Quick Start -- Remove "Next steps" from your site menu view if you have it showing. Otherwise, continue to the next step
2. Enable Quick Start from App Settings > Debug 
3. Tap "Home" on the segmented control to display the dashboard
4. ✅ A quick start card dashboard should be displayed
5. Go through all tours on both the "Customize Your Site" and the "Grow Your Audience" checklist
    -  ✅ Make sure that the tour scrolls to the correct element and/or shows a waypoint on the correct element
    -  Also, check the following cases for whether the site menu waypoint is displayed or not:

Quick Start Tour | Tour triggered from dashboard -> Shows site menu waypoint? | Tour triggered from site menu -> Shows site menu waypoint?
-- | -- | -- 
Check your site title | ✖️ | ✖️
Check your site icon | ✖️ | ✖️
Edit your home page | ✅ | ✖️
Review site pages | ✅ | ✖️
View your site| ✅ | ✖️
Social sharing | ✅ | ✖️
Publish a post | ✅ | ✖️
Follow other sites | ✖️ | ✖️
Check your site stats | ✅ | ✖️

https://user-images.githubusercontent.com/6711616/155394382-281921ad-7a24-4a68-850c-2ebc3153f098.mp4

## Regression Notes
1. Potential unintended areas of impact
- Triggering quick start tours from the Quick Start cell in BlogDetailsViewController (i.e. the site menu)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Tested manually, checked every single quick start tour from BlogDetailsViewController

3. What automated tests I added (or what prevented me from doing so)
- n/a
PR submission checklist:
